### ISSUE TITLE
A PSR request with a moved UploadedFile cannot be bridged

### ISSUE THREAD

**Issue by KorvinSzanto** [symfony/psr-http-message-bridge#15]
We call the `$uploaded_file->moveTo($path)` method when bridging a PSR request to an symfony request. This method can only be called once, if it is called more than once it throws a runtime exception breaking the bridge process.

Take this example:

``` php
<?php
$symfony_bridge->createRequest($request);
$symfony_bridge->createRequest($request);
```

If I'm not mistaken, this will fail with every PSR implementation because the second time we try to bridge to a symfony request, the uploaded files have already moved and throw `\RuntimeException`s.

Is there a way to get around this?

**Comment by KorvinSzanto** [comment]
I haven't been able to really find a solution to this, I'm not even sure that we can reliably determine if a file has been moved with PSR-7. My use case will require that requests be able to be translated 1-n times, is this not something that the bridge should be able to handle?

**Comment by KorvinSzanto** [comment]
I've updated the example. Since the file moving changes the state of the `UploadedFileInterface` implementation in a non-immutable way, simply trying to convert a psr request more than once dies.

Here's a patch for testing this case:

```diff
From dec951aa9c7e9e635edf9c6edac7fe02091b55a5 Mon Sep 17 00:00:00 2001
From: Korvin Szanto <korvinszanto@gmail.com>
Date: Mon, 9 Nov 2015 18:12:18 -0800
Subject: [PATCH] Update tests to include testing moving a PSR request twice

---
 Tests/Factory/HttpFoundationFactoryTest.php | 2 ++
 Tests/Fixtures/UploadedFile.php             | 9 ++++++++-
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/Tests/Factory/HttpFoundationFactoryTest.php b/Tests/Factory/HttpFoundationFactoryTest.php
index 1af7aec..8d63e2a 100644
--- a/Tests/Factory/HttpFoundationFactoryTest.php
+++ b/Tests/Factory/HttpFoundationFactoryTest.php
@@ -65,6 +65,7 @@ class HttpFoundationFactoryTest extends \PHPUnit_Framework_TestCase
         );
 
         $symfonyRequest = $this->factory->createRequest($serverRequest);
+        $symfonyRequest = $this->factory->createRequest($serverRequest);
         $files = $symfonyRequest->files->all();
 
         $this->assertEquals('http://les-tilleuls.coop', $symfonyRequest->query->get('url'));
@@ -212,4 +213,5 @@ class HttpFoundationFactoryTest extends \PHPUnit_Framework_TestCase
         $this->assertEquals('The response body', $symfonyResponse->getContent());
         $this->assertEquals(200, $symfonyResponse->getStatusCode());
     }
+
 }
diff --git a/Tests/Fixtures/UploadedFile.php b/Tests/Fixtures/UploadedFile.php
index 4cfa98b..9f52ef6 100644
--- a/Tests/Fixtures/UploadedFile.php
+++ b/Tests/Fixtures/UploadedFile.php
@@ -23,14 +23,16 @@ class UploadedFile implements UploadedFileInterface
     private $error;
     private $clientFileName;
     private $clientMediaType;
+    private $hasMoved;
 
-    public function __construct($filePath, $size = null, $error = UPLOAD_ERR_OK, $clientFileName = null, $clientMediaType = null)
+    public function __construct($filePath, $size = null, $error = UPLOAD_ERR_OK, $clientFileName = null, $clientMediaType = null, $hasMoved = false)
     {
         $this->filePath = $filePath;
         $this->size = $size;
         $this->error = $error;
         $this->clientFileName = $clientFileName;
         $this->clientMediaType = $clientMediaType;
+        $this->hasMoved = $hasMoved;
     }
 
     public function getStream()
@@ -40,6 +42,11 @@ class UploadedFile implements UploadedFileInterface
 
     public function moveTo($targetPath)
     {
+        if ($this->hasMoved) {
+            throw new \RuntimeException('File has already been moved.');
+        }
+
+        $this->hasMoved = true;
         rename($this->filePath, $targetPath);
     }
 
-- 
2.4.9 (Apple Git-60)

```

**Comment by nicolas-grekas** [comment]
PR welcome I suppose if anyone is willing to fix it.

**Comment by nicolas-grekas** [comment]
Fixed in #72 / v1.3.0
